GNSSLogger/app/src/main/java/com/google/android/apps/location/gps/gnsslogger/FileLogger.java

```diff
@@ -339,6 +366,25 @@ public void onNmeaReceived(long timestamp, String s) {
     }
   }
 
+  @Override
+  public void onSensorChanged(SensorEvent event) {
+    synchronized (mFileLock) {
+      if(mFileWriter == null) {
+        return;
+      }
+      mStringBuilder.setLength(0);
+      mStringBuilder.append(event.sensor.getStringType().substring(event.sensor.getStringType().lastIndexOf('.') + 1)).append(RECORD_DELIMITER);
```

Comment:
I recommend handling each sensor type separately using a series of `if else` blocks. I.e.

```
if (sensorEvent.sensor.getType() == Sensor.TYPE_ACCELEROMETER_UNCALIBRATED) { 
...
} else if ( ....
```